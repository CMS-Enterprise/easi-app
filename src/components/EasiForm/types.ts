import React from 'react';
import {
  FieldPath,
  FieldValues,
  RefCallBack,
  RegisterOptions,
  UseFormRegisterReturn,
  UseFormReturn
} from 'react-hook-form';

export type EasiPartialSubmitProps<TFieldValues extends FieldValues> = {
  update: (
    /** Object containing valid dirty field values */
    formData: Partial<TFieldValues>
  ) => Promise<any>;
  /** Callback to execute after updating */
  callback?: () => void;
  /** Whether to clear field error messages */
  clearErrors?: boolean;
};

export type EasiPartialSubmit<TFieldValues extends FieldValues> = ({
  update,
  clearErrors,
  callback
}: EasiPartialSubmitProps<TFieldValues>) => Promise<void>;

export type EasiFormRegister<TFieldValues extends FieldValues> = <
  TFieldName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>
>(
  name: TFieldName,
  options?: RegisterOptions<TFieldValues, TFieldName>
) => UseFormRegisterReturn<TFieldName> & { inputRef: RefCallBack };

export type UseEasiFormReturn<
  TFieldValues extends FieldValues = FieldValues,
  TContext = any
> = Omit<UseFormReturn<TFieldValues, TContext>, 'register'> & {
  /** Ignore errors and submit valid dirty field values */
  partialSubmit: EasiPartialSubmit<TFieldValues>;
  /** Custom register function that returns `inputRef` for function components */
  register: EasiFormRegister<TFieldValues>;
};

export type EasiFormProviderProps<
  TFieldValues extends FieldValues,
  TContext = any
> = {
  children: React.ReactNode | React.ReactNode[];
} & UseEasiFormReturn<TFieldValues, TContext>;
