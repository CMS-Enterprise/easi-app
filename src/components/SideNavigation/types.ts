export type SubNavLinkProps = {
  route: string;
  text: string;
};

export type NavLinkProps = SubNavLinkProps & {
  children?: SubNavLinkProps[];
  /** Designates end of navigation group with bottom border */
  groupEnd?: boolean;
};
