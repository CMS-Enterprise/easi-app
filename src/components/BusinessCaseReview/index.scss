.easi-pdf-export .easi-pdf-export__controls {
  margin-left: auto;
  margin-right: auto;
  max-width: 75rem;
  padding-right: 2rem;
}

.request-information-wrapper {
  padding-bottom: 1.75rem;
}
