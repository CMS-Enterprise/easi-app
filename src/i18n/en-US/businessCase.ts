const businessCase = {
  submission: {
    confirmation: {
      heading: 'Your Business Case has been submitted',
      subheading: 'Your reference ID is {{referenceId}}',
      improveEasi:
        'Help us improve EASi. <1>Tell us what you think of this service (opens in a new tab)</1>',
      homeCta: 'Go back to EASi homepage'
    }
  }
};

export default businessCase;
