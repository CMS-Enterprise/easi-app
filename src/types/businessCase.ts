import { LifecyclePhase } from 'types/estimatedLifecycle';
import { DateTime } from 'luxon';

export type EstimatedLifecycleCostLines = {
  year1: LifecyclePhase[];
  year2: LifecyclePhase[];
  year3: LifecyclePhase[];
  year4: LifecyclePhase[];
  year5: LifecyclePhase[];
};

// Base Solution
export type BusinessCaseSolution = {
  title: string;
  summary: string;
  pros: string;
  cons: string;
  estimatedLifecycleCost: EstimatedLifecycleCostLines;
  costSavings: string;
};

// Proposed Solution
export type ProposedBusinessCaseSolution = BusinessCaseSolution & {
  acquisitionApproach: string;
  hosting: {
    type: string;
    location: string;
    cloudServiceType?: string;
  };
  hasUserInterface: string;
};

type BusinessCaseStatus = 'DRAFT' | 'SUBMITTED' | 'REVIEWED' | 'REJECTED';

// Business Case Form Model
export type BusinessCaseModel = {
  id?: string;
  euaUserId?: string;
  status: BusinessCaseStatus;
  systemIntakeId: string;
  requestName: string;
  requester: {
    name: string;
    phoneNumber: string;
  };
  businessOwner: {
    name: string;
  };
  businessNeed: string;
  cmsBenefit: string;
  priorityAlignment: string;
  successIndicators: string;
  asIsSolution: BusinessCaseSolution;
  preferredSolution: ProposedBusinessCaseSolution;
  alternativeA: ProposedBusinessCaseSolution;
  alternativeB?: ProposedBusinessCaseSolution;
};

export type BusinessCasesState = {
  businessCases: BusinessCaseModel[];
  isLoading: boolean | null;
  loadedTimestamp: DateTime | null;
  error: string | null;
};

// Redux store type for business case
export type BusinessCaseState = {
  form: BusinessCaseModel;
  isLoading: boolean | null;
  isSaving: boolean;
  isSubmitting: boolean;
  error: any;
};
