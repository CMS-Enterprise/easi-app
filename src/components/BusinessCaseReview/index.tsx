import React from 'react';

import GRTFeedbackView from 'components/GRTFeedbackView';
import PDFExport from 'components/PDFExport';
import SectionWrapper from 'components/shared/SectionWrapper';
import { GetSystemIntake_systemIntake_grtFeedbacks as GRTFeedback } from 'queries/types/GetSystemIntake';
import { BusinessCaseModel } from 'types/businessCase';
import { getFiscalYear, parseAsUTC } from 'utils/date';

import AlternativeAnalysisReview from './AlternativeAnalysisReview';
import GeneralRequestInfoReview from './GeneralRequestInfoReview';
import RequestDescriptionReview from './RequestDescriptionReview';

import './index.scss';

type BusinessCaseReviewProps = {
  values: BusinessCaseModel;
  grtFeedbacks?: GRTFeedback[] | null;
  helpArticle?: boolean;
};

const BusinessCaseReview = ({
  values,
  grtFeedbacks,
  helpArticle = false
}: BusinessCaseReviewProps) => {
  const filename = `Business case for ${values.requestName}.pdf`;
  return (
    <div className="margin-top-neg-1">
      <PDFExport
        title="Business Case"
        filename={filename}
        label="Download Business Case as PDF"
        linkPosition={helpArticle ? 'top' : 'bottom'}
      >
        <div className={helpArticle ? '' : 'grid-container'}>
          <SectionWrapper
            borderBottom={!!helpArticle}
            borderTop={!!helpArticle}
            className={
              helpArticle ? 'request-information-wrapper margin-top-4' : ''
            }
          >
            <h2 className="font-heading-xl margin-top-4">
              General request information
            </h2>
            <GeneralRequestInfoReview
              values={{
                requestName: values.requestName,
                businessOwner: {
                  name: values.businessOwner.name
                },
                requester: {
                  name: values.requester.name,
                  phoneNumber: values.requester.phoneNumber
                }
              }}
            />
          </SectionWrapper>

          <SectionWrapper
            borderBottom={!!helpArticle}
            className={helpArticle ? 'padding-bottom-3' : ''}
          >
            <h2 className="font-heading-xl margin-top-4">
              Request description
            </h2>
            <RequestDescriptionReview
              values={{
                businessNeed: values.businessNeed,
                currentSolutionSummary: values.currentSolutionSummary,
                cmsBenefit: values.cmsBenefit,
                priorityAlignment: values.priorityAlignment,
                successIndicators: values.successIndicators
              }}
            />
          </SectionWrapper>
        </div>

        <div className={helpArticle ? '' : 'grid-container'}>
          <h2 className="font-heading-xl margin-top-4 easi-no-print">
            Alternatives analysis
          </h2>
        </div>
        <div className="alternative-analysis-wrapper">
          <div className={helpArticle ? '' : 'grid-container'}>
            <AlternativeAnalysisReview
              fiscalYear={
                values.createdAt
                  ? getFiscalYear(parseAsUTC(values.createdAt))
                  : new Date().getFullYear()
              }
              preferredSolution={values.preferredSolution}
              alternativeA={values.alternativeA}
              alternativeB={values.alternativeB}
            />
          </div>
        </div>
        {grtFeedbacks && grtFeedbacks.length > 0 && (
          <div className="bg-gray-10 margin-top-3 padding-x-3 padding-top-3 padding-bottom-1">
            <div className="grid-container">
              <GRTFeedbackView grtFeedbacks={grtFeedbacks} />
            </div>
          </div>
        )}
      </PDFExport>
    </div>
  );
};

export default BusinessCaseReview;
