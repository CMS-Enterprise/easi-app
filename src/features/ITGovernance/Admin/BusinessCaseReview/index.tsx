import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { GovernanceRequestFeedbackFragmentFragment } from 'gql/generated/graphql';
import { DateTime } from 'luxon';

import Alert from 'components/Alert';
import AlternativeAnalysisReview from 'components/BusinessCaseReview/AlternativeAnalysisReview';
import GeneralRequestInfoReview from 'components/BusinessCaseReview/GeneralRequestInfoReview';
import RequestDescriptionReview from 'components/BusinessCaseReview/RequestDescriptionReview';
import UswdsReactLink from 'components/LinkWrapper';
import PageHeading from 'components/PageHeading';
import PDFExport from 'components/PDFExport';
import { BusinessCaseModel } from 'types/businessCase';
import { getFiscalYear } from 'utils/date';

import ITGovAdminContext from '../../../../wrappers/ITGovAdminContext/ITGovAdminContext';

type BusinessCaseReviewProps = {
  businessCase: BusinessCaseModel;
  grtFeedbacks?: GovernanceRequestFeedbackFragmentFragment[] | null;
};

const BusinessCaseReview = ({
  businessCase,
  grtFeedbacks
}: BusinessCaseReviewProps) => {
  const { t } = useTranslation('governanceReviewTeam');
  const filename = `Business case for ${businessCase.requestName}.pdf`;

  const isITGovAdmin = useContext(ITGovAdminContext);

  if (!businessCase.id) {
    return (
      <div data-testid="business-case-review-not-found">
        <PageHeading className="margin-top-0">
          {t('general:businessCase')}
        </PageHeading>
        <p>Business Case has not been submitted</p>
      </div>
    );
  }

  return (
    <div data-testid="business-case-review">
      <PDFExport
        title="System Intake"
        filename={filename}
        label="Download Business Case as PDF"
      >
        <PageHeading className="margin-top-0">
          {t('general:businessCase')}
        </PageHeading>
        <h2 className="font-heading-xl">General request information</h2>
        <GeneralRequestInfoReview
          systemIntakeId={businessCase.systemIntakeId}
          values={{
            requestName: businessCase.requestName,
            projectAcronym: businessCase.projectAcronym,
            businessOwner: {
              name: businessCase.businessOwner.name
            },
            requester: {
              name: businessCase.requester.name,
              phoneNumber: businessCase.requester.phoneNumber
            }
          }}
        />

        <h2 className="font-heading-xl margin-top-6">Request description</h2>
        <RequestDescriptionReview
          values={{
            businessNeed: businessCase.businessNeed,
            cmsBenefit: businessCase.cmsBenefit,
            collaborationNeeded: businessCase.collaborationNeeded,
            currentSolutionSummary: businessCase.currentSolutionSummary,
            priorityAlignment: businessCase.priorityAlignment,
            successIndicators: businessCase.successIndicators,
            responseToGRTFeedback: businessCase.responseToGRTFeedback
          }}
        />
        <h2 className="font-heading-xl margin-top-6 margin-bottom-2">
          Alternatives analysis
        </h2>
        <AlternativeAnalysisReview
          fiscalYear={
            businessCase.createdAt
              ? getFiscalYear(DateTime.fromISO(businessCase.createdAt))
              : new Date().getFullYear()
          }
          preferredSolution={businessCase.preferredSolution}
          alternativeA={businessCase.alternativeA}
          alternativeB={businessCase.alternativeB}
        />

        {
          // Alert to show Feedback has moved to new tab
          grtFeedbacks && grtFeedbacks.length > 0 && (
            <Alert type="info" className="margin-top-5 easi-no-print" slim>
              {t('feedback.feedbackMoved')}
            </Alert>
          )
        }
      </PDFExport>

      {isITGovAdmin && (
        <UswdsReactLink
          className="usa-button margin-top-5"
          variant="unstyled"
          to={`/it-governance/${businessCase.systemIntakeId}/actions`}
        >
          {t('action:takeAnAction')}
        </UswdsReactLink>
      )}
    </div>
  );
};

export default BusinessCaseReview;
