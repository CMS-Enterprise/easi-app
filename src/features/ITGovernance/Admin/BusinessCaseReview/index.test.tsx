import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import {
  businessOwner,
  getSystemIntakeContactsQuery,
  systemIntake
} from 'tests/mock/systemIntake';

import { businessCaseInitialData } from 'data/businessCase';
import { BusinessCaseModel } from 'types/businessCase';
import VerboseMockedProvider from 'utils/testing/VerboseMockedProvider';

import ITGovAdminContext from '../../../../wrappers/ITGovAdminContext/ITGovAdminContext';

import BusinessCaseReview from './index';

window.matchMedia = (): any => ({
  addListener: () => {},
  removeListener: () => {}
});

describe('The GRT Business Case review', () => {
  const mockBusinessCase: BusinessCaseModel = {
    ...businessCaseInitialData,
    id: '54e829a9-6ce3-4b4b-81b0-7781b1e22821',
    systemIntakeId: systemIntake.id,
    requestName: 'Easy Access to System Information',
    requester: {
      name: 'Jane Doe',
      phoneNumber: '1234567890'
    },
    businessOwner: {
      name: businessOwner.userAccount.commonName
    },
    businessNeed: 'Mock business need',
    cmsBenefit: 'Mock CMS benefit',
    currentSolutionSummary: 'Mock current solution summary',
    priorityAlignment: 'Mock priority alignment',
    successIndicators: 'Mock success indicators',
    preferredSolution: {
      title: 'Mock Preferred solution',
      summary: 'Mock Preferred solution summary',
      acquisitionApproach: 'Mock acquisition approach',
      targetContractAwardDate: '2025-03-15T19:22:40Z',
      targetCompletionDate: '2025-03-15T19:22:40Z',
      pros: 'Mock Preferred solution pros',
      cons: 'Mock Preferred solution cons',
      estimatedLifecycleCost: {
        development: {
          label: 'Development',
          type: 'primary',
          isPresent: true,
          years: {
            year1: '1000',
            year2: '',
            year3: '',
            year4: '',
            year5: ''
          }
        },
        operationsMaintenance: {
          label: 'Operations and Maintenance',
          type: 'primary',
          isPresent: false,
          years: {
            year1: '',
            year2: '',
            year3: '',
            year4: '',
            year5: ''
          }
        },
        helpDesk: {
          label: 'Help desk/call center',
          type: 'related',
          isPresent: false,
          years: {
            year1: '',
            year2: '',
            year3: '',
            year4: '',
            year5: ''
          }
        },
        software: {
          label: 'Software licenses',
          type: 'related',
          isPresent: false,
          years: {
            year1: '',
            year2: '',
            year3: '',
            year4: '',
            year5: ''
          }
        },
        planning: {
          label: 'Planning, support, and professional services',
          type: 'related',
          isPresent: false,
          years: {
            year1: '',
            year2: '',
            year3: '',
            year4: '',
            year5: ''
          }
        },
        infrastructure: {
          label: 'Infrastructure',
          type: 'related',
          isPresent: false,
          years: {
            year1: '',
            year2: '',
            year3: '',
            year4: '',
            year5: ''
          }
        },
        oit: {
          label: 'OIT services, tools, and pilots',
          type: 'related',
          isPresent: false,
          years: {
            year1: '',
            year2: '',
            year3: '',
            year4: '',
            year5: ''
          }
        },
        other: {
          label: 'Other services, tools, and pilots',
          type: 'related',
          isPresent: false,
          years: {
            year1: '',
            year2: '',
            year3: '',
            year4: '',
            year5: ''
          }
        }
      },
      costSavings: 'Mock cost savings',
      security: {
        isApproved: false,
        isBeingReviewed: 'YES'
      },
      zeroTrustAlignment: 'Mock Zero Trust',
      hosting: {
        type: 'Cloud',
        location: 'Mock location',
        cloudStrategy: 'Mock cloud strategy',
        cloudServiceType: 'Mock cloud service'
      },
      hasUserInterface: 'YES',
      workforceTrainingReqs: 'Mock workforce training requirements'
    }
  };

  it('matches the snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <VerboseMockedProvider mocks={[getSystemIntakeContactsQuery()]}>
          <BusinessCaseReview businessCase={mockBusinessCase} />
        </VerboseMockedProvider>
      </MemoryRouter>
    );

    expect(
      await screen.findByText(businessOwner.userAccount.commonName)
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders action button for GRT admins', async () => {
    render(
      <MemoryRouter>
        <VerboseMockedProvider mocks={[getSystemIntakeContactsQuery()]}>
          <ITGovAdminContext.Provider value>
            <BusinessCaseReview businessCase={mockBusinessCase} />
          </ITGovAdminContext.Provider>
        </VerboseMockedProvider>
      </MemoryRouter>
    );

    expect(
      await screen.findByText(businessOwner.userAccount.commonName)
    ).toBeInTheDocument();

    expect(
      screen.getByRole('link', { name: 'Take an action' })
    ).toBeInTheDocument();
  });
});
