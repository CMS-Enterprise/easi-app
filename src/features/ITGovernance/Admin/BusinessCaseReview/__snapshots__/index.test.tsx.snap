// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The GRT Business Case review > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="business-case-review"
  >
    <div
      class="easi-pdf-export"
    >
      <h1
        class="easi-only-print"
      >
        System Intake
      </h1>
      <h1
        aria-live="polite"
        class="easi-h1 margin-top-0"
        tabindex="-1"
      >
        Business Case
      </h1>
      <h2
        class="font-heading-xl"
      >
        General request information
      </h2>
      <dl
        title="General request information"
      >
        <div
          class="easi-review-row"
        >
          <div>
            <dt
              class="text-bold margin-bottom-05"
            >
              Contact / request title
            </dt>
            <dd
              class="description-definition margin-0"
            >
              Easy Access to System Information
            </dd>
          </div>
          <div>
            <dt
              class="text-bold margin-bottom-05"
            >
              Contact / request acronym
            </dt>
            <dd
              class="description-definition margin-0"
            />
          </div>
        </div>
        <div
          class="easi-review-row"
        >
          <div>
            <dt
              class="text-bold margin-bottom-05"
            >
              CMS Business Owner
            </dt>
            <dd
              class="description-definition margin-0"
            >
              Jane Doe
            </dd>
          </div>
        </div>
        <div
          class="easi-review-row"
        >
          <div>
            <dt
              class="text-bold margin-bottom-05"
            >
              Requester
            </dt>
            <dd
              class="description-definition margin-0"
            >
              Jane Doe
            </dd>
          </div>
        </div>
        <div
          class="easi-review-row"
        >
          <div>
            <dt
              class="text-bold margin-bottom-05"
            >
              Requester phone number
            </dt>
            <dd
              class="description-definition margin-0"
            >
              1234567890
            </dd>
          </div>
        </div>
      </dl>
      <h2
        class="font-heading-xl margin-top-6"
      >
        Request description
      </h2>
      <dl
        title="Request description"
      >
        <div
          class="easi-review-row margin-bottom-3"
        >
          <div
            class="line-height-body-3"
          >
            <dt
              class="text-bold margin-bottom-05"
            >
              What is your business or user need?
            </dt>
            <dd
              class="description-definition margin-0 text-pre-wrap"
            >
              Mock business need
            </dd>
          </div>
        </div>
        <div
          class="easi-review-row margin-bottom-3"
        >
          <div
            class="line-height-body-3"
          >
            <dt
              class="text-bold margin-bottom-05"
            >
              What internal collaboration or vendor engagement will support the recommended work?
            </dt>
            <dd
              class="description-definition margin-0 text-pre-wrap"
            />
          </div>
        </div>
        <div
          class="easi-review-row margin-bottom-3"
        >
          <div
            class="line-height-body-3"
          >
            <dt
              class="text-bold margin-bottom-05"
            >
              What is the current process that you are using for this work, or how is this need currently being met?
            </dt>
            <dd
              class="description-definition margin-0 text-pre-wrap"
            >
              Mock current solution summary
            </dd>
          </div>
        </div>
        <div
          class="easi-review-row margin-bottom-3"
        >
          <div
            class="line-height-body-3"
          >
            <dt
              class="text-bold margin-bottom-05"
            >
              How will CMS benefit from this effort?
            </dt>
            <dd
              class="description-definition margin-0 text-pre-wrap"
            >
              Mock CMS benefit
            </dd>
          </div>
        </div>
        <div
          class="easi-review-row margin-bottom-3"
        >
          <div
            class="line-height-body-3"
          >
            <dt
              class="text-bold margin-bottom-05"
            >
              How does this effort align with organizational priorities?
            </dt>
            <dd
              class="description-definition margin-0 text-pre-wrap"
            >
              Mock priority alignment
            </dd>
          </div>
        </div>
        <div
          class="easi-review-row margin-bottom-3"
        >
          <div
            class="line-height-body-3"
          >
            <dt
              class="text-bold margin-bottom-05"
            >
              How will you determine whether or not this effort is successful?
            </dt>
            <dd
              class="description-definition margin-0 text-pre-wrap"
            >
              Mock success indicators
            </dd>
          </div>
        </div>
        <div
          class="easi-review-row margin-bottom-3"
        >
          <div
            class="line-height-body-3"
          >
            <dt
              class="text-bold margin-bottom-05"
            >
              How will you implement or respond to the recommendations provided by the GRT?
            </dt>
            <dd
              class="description-definition margin-0 text-pre-wrap"
            >
              N/A
            </dd>
          </div>
        </div>
      </dl>
      <h2
        class="font-heading-xl margin-top-6 margin-bottom-2"
      >
        Alternatives analysis
      </h2>
      <div
        class="easi-responsive-tabs bg-base-lightest"
      >
        <div
          class="easi-responsive-tabs__navigation"
        >
          <div
            class="easi-responsive-tabs__tabs-wrapper"
          >
            <ul
              class="easi-responsive-tabs__tab-list"
            >
              <li
                class="easi-responsive-tabs__tab easi-responsive-tabs__tab--selected"
              >
                <button
                  class="easi-responsive-tabs__tab-btn"
                  type="button"
                >
                  <span
                    class="easi-responsive-tabs__tab-divider"
                  >
                    <span
                      class="easi-responsive-tabs__tab-text"
                    >
                      Preferred solution
                    </span>
                  </span>
                </button>
              </li>
            </ul>
          </div>
          <div />
        </div>
        <div
          class="bg-white easi-business-case__review-solutions-wrapper"
          style="overflow: auto;"
        >
          <h3
            class="easi-only-print business-case-solution-header"
          >
            Preferred solution
          </h3>
          <dl
            class="margin-bottom-0"
            title="Preferred solution"
          >
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Solution Title
                </dt>
                <dd
                  class="description-definition margin-0"
                >
                  Mock Preferred solution
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Summary
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Mock Preferred solution summary
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Acquisition approach
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Mock acquisition approach
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Target release date of contract solicitation
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  03/15/2025
                </dd>
              </div>
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Target completion date (In production)
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  03/15/2025
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Is your solution approved by IT Security for use at CMS (FedRAMP, FISMA approved, within the CMS cloud enclave)?
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  No
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Is it in the process of getting approved?
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Yes
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Identify your project's alignment with Zero Trust principles?
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Mock Zero Trust
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Do you need to host your solution?
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                />
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Will your solution have a User Interface, be public facing, or involve outside customers?
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Yes
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Will any workforce training requirements arise as a result of this solution?
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Mock workforce training requirements
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Pros
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Mock Preferred solution pros
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  Cons
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Mock Preferred solution cons
                </dd>
              </div>
            </div>
            <div
              class="easi-review-row"
            >
              <div
                class="margin-y-1"
              >
                <div
                  class="est-lifecycle-cost__total bg-base-lightest overflow-auto padding-x-2"
                >
                  <dl
                    title="System total cost"
                  >
                    <dt
                      class="text-bold margin-bottom-05"
                    >
                      System total cost
                    </dt>
                    <dd
                      class="description-definition margin-0"
                    >
                      $1,000
                    </dd>
                  </dl>
                </div>
              </div>
            </div>
            <div
              class="easi-review-row margin-bottom-0"
            >
              <div
                class="line-height-body-3"
              >
                <dt
                  class="text-bold margin-bottom-05"
                >
                  What is the cost savings or avoidance associated with this solution?
                </dt>
                <dd
                  class="description-definition margin-0 text-pre-wrap"
                >
                  Mock cost savings
                </dd>
              </div>
            </div>
          </dl>
        </div>
      </div>
    </div>
    <div
      class="easi-pdf-export__controls margin-top-6"
    >
      <button
        class="usa-button usa-button--unstyled display-flex flex-align-center easi-no-print"
        data-testid="button"
        type="button"
      >
        <svg
          class="usa-icon margin-right-1"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
          />
        </svg>
        Download Business Case as PDF
      </button>
    </div>
  </div>
</DocumentFragment>
`;
